client/my-sites/add-ons/components/storage-add-ons-card.tsx

```diff
@@ -211,10 +208,7 @@ export default function StorageAddOnCard( { siteId, actionPrimary }: Props ) {
 	}
 
 	const onActionPrimary = () => {
-		actionPrimary?.handler(
-			availableStorageAddOns[ 0 ].productSlug,
-			selectedStorageAddOnStorageQuantity
-		);
+		actionPrimary?.( availableStorageAddOns[ 0 ].productSlug, selectedStorageAddOnStorageQuantity );
```

Comment:
Looking at adjacent code, we are using the optional chaining operator liberally, expecting nullish values. 

I recognize this is pre-existing but noting that we are not doing so here. Re: `availableStorageAddOns[ 0 ].productSlug`

Reply:
Sorry, could you expand what you meant? Which adjacent code and what are we not doing here? 🤔

Reply:
I guess it's related to the following line that uses `availableStorageAddOns?.map`

```js
const selectControlOptions = availableStorageAddOns?.map( ( addOn ) => {
```

After taking a closer look, the `useAvailableStorageAddOns` hook should always return an array, though it might be empty. So we should be careful when accessing `availableStorageAddOns[0].productSlug` directly. To avoid potential issues, it's better to use optional chaining like `availableStorageAddOns[0]?.productSlug`.

Reply:
Got it, it's fixed in the latest commits.

Reply:
Thanks @arthur791004. I'll be more thorough next time. :)